modules/KoreBuild.Tasks/GenerateDependenciesPropsFile.cs

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using KoreBuild.Tasks.ProjectModel;
+using Microsoft.Build.Construction;
+using Microsoft.Build.Evaluation;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace KoreBuild.Tasks
+{
+    /// <summary>
+    /// Generates the build/dependencies.props file base on your current project,
+    /// and updates all PackageReferences in <see cref="Projects"/>  and <see cref="OtherImports"/> to use MSBuild variables.
+    /// </summary>
+    public class GenerateDependenciesPropsFile : Microsoft.Build.Utilities.Task, ICancelableTask
+    {
+        private readonly CancellationTokenSource _cts = new CancellationTokenSource();
+
+        /// <summary>
+        /// The projects to update
+        /// </summary>
+        [Required]
+        public ITaskItem[] Projects { get; set; }
+
+        /// <summary>
+        /// other files to update that may have PackageReferences in them
+        /// </summary>
+        public ITaskItem[] OtherImports { get; set; }
+
+        /// <summary>
+        /// Location of the dependenices.props file
+        /// </summary>
+        [Required]
+        public string DependenciesFile { get; set; }
+
+        /// <summary>
+        /// Additional properties to use when evaluating the projects
+        /// </summary>
+        public string[] Properties { get; set; }
+
+        public void Cancel()
+        {
+            _cts.Cancel();
+        }
+
+        public override bool Execute()
+        {
+            var unifiedPackageList = new Dictionary<string, PackageReferenceInfo>(StringComparer.OrdinalIgnoreCase);
+
+            var projects = new ProjectInfoFactory(Log).CreateMany(Projects, Properties, false, _cts.Token);
+            var packageRefs = projects.SelectMany(p => p.Frameworks).SelectMany(f => f.Dependencies);
+
+            foreach (var packageRef in packageRefs)
+            {
+                if (unifiedPackageList.TryGetValue(packageRef.Value.Id, out var other))
+                {
+                    if (other.Version != packageRef.Value.Version)
+                    {
+                        Log.LogError($"Conflicting dependency versions for {packageRef.Value.Id}: {other.Project.FileName} references v{other.Version} but {packageRef.Value.Project.FileName} references {packageRef.Value.Version}");
```

Comment:
Good question. If a solution has multiple versions of the same package ID, this is either an error in the project setup, or its an edge case that we should consider carefully.

Today, I can only think of one valid reason to have multiple versions of the same package ID: EF Core's benchmark projects, where they profile against previous releases of EF Core. My recommendation for this case is that the benchmarks repo should not build as a part of the regular build. It can be a separate target for the repo.